### DISCUSSION TITLE
4.0.0 插件依赖顺序？

### DISCUSSION THREAD

**Discussion by laggardkernel** [IrineSistiana/mosdns#398] [General]
现在必须先声明 tag 再使用，变得严格依赖定义顺序。也是因为更换了解析库？不能做原来那种变量提升了吗？

**Comment by IrineSistiana** [comment]
以前是动态解释，每次运行都会根据tag找插件。现在是静态的，初始化时逻辑就已经固定了。

> 变量提升

高端了。解析代码是自己写的，不想太麻烦。

**Comment by laggardkernel** [comment]
> 以前是动态解释，每次运行都会根据tag找插件。现在是静态的，初始化时逻辑就已经固定了。

原来如此。
